src/api/libcellml/model.h

```diff
@@ -366,6 +366,89 @@ class LIBCELLML_EXPORT Model: public ComponentEntity
      */
     bool fixVariableInterfaces();
 
+    /**
+     * @brief Add an import source item to this model.
+     *
+     * Add import source by reference to the model. If the import
+     * source item was previously in a different model, it is moved
+     * to this one, and the previous model's import source list is updated.
+     *
+     * The function will return @c false and no action is taken if:
+     *  - The @p imp pointer already exists in this model; or
+     *  - The @p imp is @c nullptr.
+     *
+     * @param imp The import source to add.
+     *
+     * @return @c true if the import source item is added or @c false otherwise.
+     */
+    bool addImportSource(const ImportSourcePtr &imp);
+
+    /**
+     * @brief Get the number of import source items in the model.
+     *
+     * Returns the number of import source items the model contains.
+     *
+     * @return The number of import source items.
+     */
+    size_t importSourceCount() const;
+
+    /**
+     * @brief Get the import source item at the given @p index.
+     *
+     * Returns a reference to an import source at the given @p index.  If the @p index
+     * is not valid a @c nullptr is returned, the range of valid values for the
+     * index is [0, \#import sources).
```

Comment:
I would replace wih "#importSources".

Reply:
Same elsewhere.

Reply:
I don't understand the reason.  Elsewhere we use lowercase descriptions? It would be better to ditch the `#` altogether as it's weird to use this kind of shorthand here.  Better to spell it out properly, or provide a reference to the `importSourceCount()` function.

Reply:
Well, we do have "[0, \#equivalentVariables)" in `variable.h`. I am "fine" with whatever we go for, as long as we are being consistent and here we are not.

Reply:
7a544255